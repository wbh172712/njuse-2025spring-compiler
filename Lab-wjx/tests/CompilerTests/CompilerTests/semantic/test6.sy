int main(int argc, int argv[][3]) {
  int array[2] = {1, 2};
  int a[3] = -array;
  return 0;
}

int f(int k, int k) {
  return main(-main, 3);
}

void test() {
  int f;
  int b = f(f);
  int a[3][3][3];
  int c[2][2];
  a[1] = c;
  a[1][1] = c;
}

void test2() {
  int a = {1, 2};
  {
    int b = 3;
  }
  int b = 4;
  if (c) {
    b = 5;
  } else {
    b = 6;
  }
  int c[3];
  c[1][3] = 2;
  if (c > 7) {
    f();
  }
}