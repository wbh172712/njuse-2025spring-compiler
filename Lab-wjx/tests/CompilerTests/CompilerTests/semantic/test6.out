Error type 6 at Line 3: type.Type mismatched for operands.
Error type 3 at Line 7: Redefined variable: k.
Error type 6 at Line 8: type.Type mismatched for operands.
Error type 8 at Line 8: Function is not applicable for arguments.
Error type 8 at Line 8: Function is not applicable for arguments.
Error type 10 at Line 13: Not a function: f.
Error type 5 at Line 17: type.Type mismatched for assignment.
Error type 5 at Line 21: type.Type mismatched for assignment.
Error type 1 at Line 26: Undefined variable: c.
Error type 9 at Line 32: Not an array: c[].
Error type 6 at Line 33: type.Type mismatched for operands.
Error type 8 at Line 34: Function is not applicable for arguments.